### ISSUE TITLE
MWC canary fails type checking

### ISSUE THREAD

**Issue by aomarks** [material-components/material-web#2715]
repro:

```sh
mkdir repro
cd repro
npm init --yes
npm i typescript @material/mwc-button@0.24.0-canary.cd9110cd.0
echo 'import "@material/mwc-button";' > index.ts
echo '{"compilerOptions":{"moduleResolution":"node","module":"esnext","target":"es2020","lib":["esnext","DOM"]}}' > tsconfig.json
npx tsc
```

tsc error:
```
node_modules/@material/mwc-button/mwc-button-base.d.ts:38:130 - error TS2694: Namespace '"/Users/aomarks/scratch/mwc/repro/node_modules/lit-html/directives/class-map"' has no exported member 'ClassMapDirective'.

38     protected getRenderClasses(): import("lit-html/directive").DirectiveResult<typeof import("lit-html/directives/class-map.js").ClassMapDirective>;
                                                                                                                                    ~~~~~~~~~~~~~~~~~
```

**Comment by dfreedm** [comment]
What version of node are you using?

**Comment by aomarks** [comment]
Tried these 3, all the same error:

- node v16.9.1 / npm 7.21.1
- node v15.8.0 / npm 7.7.3
- node v14.17.6 / npm 6.14.15

**Comment by dfreedm** [comment]
I assume this is Typescript 4.4? I think @e111077 mentioned that this generated typing is super funky.
We [don't have a return type on that function](https://github.com/material-components/material-web/blob/master/packages/button/mwc-button-base.ts#L92), so I'm not sure what's going on in Typescript to generate that weird type statement.

**Comment by aomarks** [comment]
> I assume this is Typescript 4.4?

I get the same error using TypeScript 4.4.3, 4.3.5, and 4.2.4.

**Comment by aomarks** [comment]
Update, this appears to be a TypeScript bug relating to the use of type-only exports in `lit`. I filed an issue at https://github.com/microsoft/TypeScript/issues/45946, and am currently investigating a workaround that we could ship in the next `lit` release.

**Comment by aomarks** [comment]
Verified this is now fixed in the latest canary `cd9110cd`:

```sh
mkdir repro
cd repro
npm init --yes
npm i typescript @material/mwc-button@0.24.0-canary.cd9110cd.0
echo 'import "@material/mwc-button";' > index.ts
echo '{"compilerOptions":{"moduleResolution":"node","module":"esnext","target":"es2020","lib":["esnext","DOM"]}}' > tsconfig.json
npx tsc
```
